rasa/core/channels/slack.py

```diff
@@ -268,21 +268,23 @@ def _sanitize_user_message(
         uids_to_remove = uids_to_remove or []
 
         for uid_to_remove in uids_to_remove:
+            escaped_uid = re.escape(uid_to_remove)
+
             # heuristic to format majority cases OK
             # can be adjusted to taste later if needed,
             # but is a good first approximation
             for regex, replacement in [
-                (rf"<@{uid_to_remove}>\s", ""),
-                (rf"\s<@{uid_to_remove}>", ""),  # a bit arbitrary but probably OK
-                (rf"<@{uid_to_remove}>", " "),
+                (rf"<@{escaped_uid}>\s", ""),
+                (rf"\s<@{escaped_uid}>", ""),  # a bit arbitrary but probably OK
+                (rf"<@{escaped_uid}>", " "),
             ]:
                 text = re.sub(regex, replacement, text)
 
-        # Find multiple mailto or http links like
+        # Find multiple case-insensitive mailto or http links like
```

Comment:
Why did we need to change the pattern for finding case-insensitive links?

Reply:
This change wasn't related to the security vulnerability. I just thought that a case-sensitive regex looks like a bug so I fixed it. Do you think I should isolate this change to a different ticket/PR for transparency?

Reply:
(the previous regex was case sensitive, it only works for URLs in lowercase)

Reply:
Yeah, I'd look in Jira if there has been a request for changing this before.
Also isolating would be a good idea to see which one of the two changes causes the unit test to fail.

Reply:
I've removed the case-insensitivity from the regex now